packages/robotics/pitop/robotics/drive_controller.py

```diff
@@ -156,45 +202,73 @@ def right(self, speed_factor, turn_radius=0):
         :param float turn_radius: Radius used by the robot to perform
             the movement. Using `turn_radius=0` will cause the robot to
             rotate in place.
+        :param float distance: Distance to travel in meters. If not
+            provided, the robot will move indefinitely
         """
 
-        self.left(-speed_factor, -turn_radius)
+        self.left(-speed_factor, -turn_radius, distance)
 
-    def target_lock_drive_angle(self, angle):
+    def target_lock_drive_angle(self, angle: Union[int, float]) -> None:
         """Make the robot move in the direction of the specified angle, while
         maintaining the current linear speed."""
         angular_speed = self.__target_lock_pid_controller(angle)
         self.robot_move(self._linear_speed_x_hold, angular_speed)
 
-    def rotate(self, angle, time_to_take):
+    def rotate(
+        self,
+        angle: Union[int, float],
+        time_to_take: Optional[Union[int, float]] = None,
+        force_time_to_take: bool = False,
```

Comment:
```suggestion
        max_speed_factor: float = 0.3,
```

When max_speed_factor is above 0.3 give a warning saying rotation may be inconsistent.